managedplugin/download.go

```diff
@@ -176,6 +163,42 @@ func DownloadPluginFromHub(ctx context.Context, authToken, localPath, team, name
 	return nil
 }
 
+func downloadPluginAssetFromHub(ctx context.Context, authToken, teamName, pluginTeam, name, version string, typ PluginType) (*cloudquery_api.PluginAsset, int, error) {
+	var pluginAsset *cloudquery_api.PluginAsset
+	var statusCode int
+
+	c, err := cloudquery_api.NewClientWithResponses(APIBaseURL(),
+		cloudquery_api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+			if authToken != "" {
+				req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", authToken))
+			}
+			return nil
+		}))
+	if err != nil {
+		return pluginAsset, statusCode, fmt.Errorf("failed to create Hub API client: %w", err)
+	}
+
+	target := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+
+	aj := "application/json"
+
+	if teamName != "" {
+		resp, err := c.DownloadPluginAssetByTeamWithResponse(ctx, teamName, pluginTeam, cloudquery_api.PluginKind(typ.String()), name, version, target, &cloudquery_api.DownloadPluginAssetByTeamParams{Accept: &aj})
+		if err != nil {
+			return pluginAsset, statusCode, fmt.Errorf("failed to get plugin url: %w", err)
+		}
+		pluginAsset, statusCode = resp.JSON200, resp.StatusCode()
+	} else {
+		resp, err := c.DownloadPluginAssetWithResponse(ctx, pluginTeam, cloudquery_api.PluginKind(typ.String()), name, version, target, &cloudquery_api.DownloadPluginAssetParams{Accept: &aj})
+		if err != nil {
+			return pluginAsset, statusCode, fmt.Errorf("failed to get plugin url: %w", err)
+		}
+		pluginAsset, statusCode = resp.JSON200, resp.StatusCode()
+	}
+
+	return pluginAsset, statusCode, nil
```

Comment:
can be written in a more compact way:

no need for `var statusCode`, teamname check can be a switch case with `return resp.JSON200, resp.StatusCode()`

also maybe separate error messages for `failed to get plugin url` so that we know which branch it got into?

Reply:
Addressed in https://github.com/cloudquery/plugin-pb-go/pull/162/commits/2eeea5e5eace640b7b993be6a0f4fabd44195796

Generally (maybe Java mindset) I don't use switch statements if there are < 3 options.